src/scripts/main.js

```diff
@@ -1,3 +1,224 @@
 'use strict';
 
-// write your code here
+const button = document.querySelector('button');
+const messageStart = document.querySelector('.message-start');
+const messageWin = document.querySelector('.message-win');
+const messageLose = document.querySelector('.message-lose');
+const gameScore = document.querySelector('.game-score');
+const fieldRows = document.querySelectorAll('.field-row');
+
+const size = 4;
+let score = 0;
+let isWin = false;
+let newGameField;
+let gameField = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+button.addEventListener('click', e => {
+  document.addEventListener('keydown', move);
+
+  if (button.classList.contains('start')) {
+    button.classList.replace('start', 'restart');
+    button.innerText = 'Restart';
+    messageStart.classList.add('hidden');
+  } else {
+    isWin = false;
+    reset();
+  }
+
+  addNumber();
+  addNumber();
+  render();
+});
+
+function move(e) {
+  newGameField = gameField;
```

Comment:
You are assigning the reference of gameField to newGameField, not creating a new copy. This may cause unexpected behavior as changes to newGameField will affect gameField. Instead, use a method like slice or the spread operator to create a new copy of the array.